src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -313,6 +317,41 @@ protected override int RunLocalCompilation(string[] arguments, BuildPaths buildP
             }
         }
 
+        // Protected for testing
+        protected static bool TryConnectToNamedPipeWithSpinWait(NamedPipeClientStream pipeStream,
+                                                                int timeoutMs,
+                                                                CancellationToken cancellationToken)
+        {
+            // .NET 4.5 implementation of NamedPipeStream.Connect busy waits the entire time.
+            // Work around is to spin wait.
+            const int maxWaitInterval = 50;
+            int elapsed = 0;
+            var sw = new SpinWait();
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                int waitTime = Math.Min(timeoutMs - elapsed, maxWaitInterval);
+                if (timeoutMs != Timeout.Infinite && waitTime <= 0)
+                {
+                    return false;
+                }
+
+                try
+                {
+                    pipeStream.Connect(waitTime);
+                    break;
+                }
+                catch (Exception e) when (e is IOException || e is TimeoutException)
+                {
+                    // Ignore timeout
```

Comment:
It is not clear why are we ignoring IOException. Perhaps explain in the comment?
Also, would it be better to have two strongly typed catch clauses instead of using a filter?